### ISSUE TITLE
Can't use mongodb rust driver without compiler claims trait-related issues (E0277)

### ISSUE THREAD

**Issue by jolmedo** [mongodb/mongo-rust-driver#731]
## Versions/Environment
1. Rust versi0n: 1.63
2. Ubuntu 20.04
3. 
https://github.com/rust-lang/crates.io-index#mongodb@2.3.0
https://github.com/rust-lang/crates.io-index#bson@2.4.0

4. What version of MongoDB are you using? (Check with the MongoDB shell using `db.version()`) 
5.0.4
5. What is your MongoDB topology (standalone, replica set, sharded cluster, serverless)?
replica set

## Describe the bug
At compile time, error E0277 arises with the following message: the trait bound `ValueBag<'_>: From<u128>` is not satisfied. The code pretends to get the list of databases hosted in the MongoDB instance and print'em.
I'm pretty new at Rust, and a solid consumer of MongoDB ecosystem, via mongosh, via other APIs like the ones from Python, Java, Scala, etc. And I'm also pretty sure that I run this code using Rust 1.57. 

To reproduce the code, just type one of the examples:

```
use mongodb::{Client, options::ClientOptions};

async fn main() -> mongodb::error::Result<()>{
   let mut client_options = ClientOptions::parse(
    "mongodb://<a_member_of_ReplSet_IP>,<another_member_of_ReplSet_IP/?replicaSet=myReplSet").await?;
    client_options.app_name = Some("My Copier app".to_string());
    let client = Client::with_options(client_options)?;
    for db_name in client.list_database_names(None, None).await? {
        println!("{}", db_name);
    }
    Ok(())
}
```

The error occurs at compile time:

```
Compiling log v0.4.14
error[E0277]: the trait bound `ValueBag<'_>: From<u128>` is not satisfied
   --> /home/jolmedo/.cargo/registry/src/github.com-1ecc6299db9ec823/log-0.4.14/src/kv/value.rs:364:21
    |
364 |                       Value::from_value_bag(value)
    |                       ^^^^^^^^^^^^^^^^^^^^^ the trait `From<u128>` is not implemented for `ValueBag<'_>`
...
384 | / impl_to_value_primitive![
385 | |     usize, u8, u16, u32, u64, u128, isize, i8, i16, i32, i64, i128, f32, f64, char, bool,
386 | | ];
    | |_- in this macro invocation
    |
    = help: the following other types implement trait `From<T>`:
              <ValueBag<'v> as From<&'a ()>>
              <ValueBag<'v> as From<&'a bool>>
              <ValueBag<'v> as From<&'a char>>
              <ValueBag<'v> as From<&'a f32>>
              <ValueBag<'v> as From<&'a f64>>
              <ValueBag<'v> as From<&'a i16>>
              <ValueBag<'v> as From<&'a i32>>
              <ValueBag<'v> as From<&'a i64>>
            and 25 others
    = note: required because of the requirements on the impl of `Into<ValueBag<'_>>` for `u128`
note: required by a bound in `Value::<'v>::from_value_bag`
   --> /home/jolmedo/.cargo/registry/src/github.com-1ecc6299db9ec823/log-0.4.14/src/kv/value.rs:250:12
    |
248 |     fn from_value_bag<T>(value: T) -> Self
    |        -------------- required by a bound in this
249 |     where
250 |         T: Into<ValueBag<'v>>,
    |            ^^^^^^^^^^^^^^^^^^ required by this bound in `Value::<'v>::from_value_bag`
    = note: this error originates in the macro `impl_to_value_primitive` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `ValueBag<'_>: From<i128>` is not satisfied
   --> /home/jolmedo/.cargo/registry/src/github.com-1ecc6299db9ec823/log-0.4.14/src/kv/value.rs:364:21
    |
364 |                       Value::from_value_bag(value)
    |                       ^^^^^^^^^^^^^^^^^^^^^ the trait `From<i128>` is not implemented for `ValueBag<'_>`
...
384 | / impl_to_value_primitive![
385 | |     usize, u8, u16, u32, u64, u128, isize, i8, i16, i32, i64, i128, f32, f64, char, bool,
386 | | ];
    | |_- in this macro invocation
    |
    = help: the following other types implement trait `From<T>`:
              <ValueBag<'v> as From<&'a ()>>
              <ValueBag<'v> as From<&'a bool>>
              <ValueBag<'v> as From<&'a char>>
              <ValueBag<'v> as From<&'a f32>>
              <ValueBag<'v> as From<&'a f64>>
              <ValueBag<'v> as From<&'a i16>>
              <ValueBag<'v> as From<&'a i32>>
              <ValueBag<'v> as From<&'a i64>>
            and 25 others
    = note: required because of the requirements on the impl of `Into<ValueBag<'_>>` for `i128`
note: required by a bound in `Value::<'v>::from_value_bag`
   --> /home/jolmedo/.cargo/registry/src/github.com-1ecc6299db9ec823/log-0.4.14/src/kv/value.rs:250:12
    |
248 |     fn from_value_bag<T>(value: T) -> Self
    |        -------------- required by a bound in this
249 |     where
250 |         T: Into<ValueBag<'v>>,
    |            ^^^^^^^^^^^^^^^^^^ required by this bound in `Value::<'v>::from_value_bag`
    = note: this error originates in the macro `impl_to_value_primitive` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `log` due to 2 previous errors

Process finished with exit code 101
```
Thanks in advance, for time and concern.
Best regards.

**Comment by kmahar** [comment]
Hi @jolmedo, glad to see you ended up solving your issue; what did the fix end up being? I found this similar discussion here that suggested it might have to do with conflicting `log` versions dependencies use: https://github.com/SeaQL/sea-orm/discussions/692

**Comment by jolmedo** [comment]
Hi @kmahar , it was exactly as you wrote, an issue with the ``log`` versions: version I was using, by default, was 0.4.14, I started to use v0.4.17 but issue was fixed since v0.4.16. I hope this could help everybody.

**Comment by kmahar** [comment]
cool, thank you for following up!!
